### PR TITLE
new ZED category field "body-class"

### PR DESCRIPTION
**Pull request by DerCalli**
**Changelog**

- add zed category field body class
- propel schema
- transfer
- add missing unit-tests

### CODE REVIEW

**Review by julianhyatt — APPROVED** [review]

File: tests/FondOfSpryker/Zed/Communication/Plugin/BodyClassTypePluginTest.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+     * @var \FondOfSpryker\Zed\CategoryFieldExtender\Communication\Form\BodyClassType|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected $bodyClassTypeMock;
+
+    /**
+     * @return void
+     */
+    protected function _before()
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Plugin/BodyClassTypePluginTest.php]: Remove comment...

File: tests/FondOfSpryker/Zed/Communication/Form/LocalizedAttributesCollectionTypeTest.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+     * @var \Symfony\Component\Form\AbstractType
+     */
+    protected $type;
+
+    /**
+     * @return void
+     */
+    protected function _before(): void
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Form/LocalizedAttributesCollectionTypeTest.php]: Remove comment...

File: tests/FondOfSpryker/Zed/Communication/Form/ContentfulHeroTypeTest.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+     * @var \Symfony\Component\Form\AbstractType
+     */
+    protected $type;
+
+    /**
+     * @return void
+     */
+    protected function _before(): void
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Form/ContentfulHeroTypeTest.php]: Remove comment...

File: tests/FondOfSpryker/Zed/Communication/Form/ContentfulCollectionListingTypeTest.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+     * @var \Symfony\Component\Form\AbstractType
+     */
+    protected $type;
+
+    /**
+     * @return void
+     */
+    protected function _before(): void
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Form/ContentfulCollectionListingTypeTest.php]: Remove comment

File: tests/FondOfSpryker/Zed/Communication/Form/ContentfulCollectionBetweenListingTypeTest.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+     * @var \Symfony\Component\Form\AbstractType
+     */
+    protected $type;
+
+    /**
+     * @return void
+     */
+    protected function _before(): void
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Form/ContentfulCollectionBetweenListingTypeTest.php]: Remove comment...

**Review by sourcecube — APPROVED** [review]
I think those tests should test that the given type was added. Just testing that the class calls an `add` function doesn't bring any more information that the component works as expected.

File: tests/FondOfSpryker/Zed/Communication/Form/ContentfulCollectionBeforeListingTypeTest.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+     * @var \Symfony\Component\Form\AbstractType
+     */
+    protected $type;
+
+    /**
+     * @return void
+     */
+    protected function _before(): void
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Form/ContentfulCollectionBeforeListingTypeTest.php]: Remove comment

File: tests/FondOfSpryker/Zed/Communication/Form/ContentfulCollectionAfterListingTypeTest.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+     * @var \Symfony\Component\Form\AbstractType
+     */
+    protected $type;
+
+    /**
+     * @return void
+     */
+    protected function _before(): void
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Form/ContentfulCollectionAfterListingTypeTest.php]: Remove comment...

File: tests/FondOfSpryker/Zed/Communication/Form/CategoryContentTypeTypeTest.php [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+     * @var \Symfony\Component\Form\AbstractType
+     */
+    protected $type;
+
+    /**
+     * @return void
+     */
+    protected function _before(): void
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Form/CategoryContentTypeTypeTest.php]: Remove comment...

File: tests/FondOfSpryker/Zed/Communication/Form/BodyClassTypeTest.php [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+     * @var \Symfony\Component\Form\AbstractType
+     */
+    protected $type;
+
+    /**
+     * @return void
+     */
+    protected function _before(): void
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Form/BodyClassTypeTest.php]: Remove comment...

File: tests/FondOfSpryker/Zed/Communication/Form/AltTitleTypeTest.php [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+     * @var \Symfony\Component\Form\AbstractType
+     */
+    protected $type;
+
+    /**
+     * @return void
+     */
+    protected function _before(): void
+    {
+        parent::_before(); // TODO: Change the autogenerated stub
```
Comment (daniel-rose) [inline tests/FondOfSpryker/Zed/Communication/Form/AltTitleTypeTest.php]: Remove comment...

**Review by daniel-rose — APPROVED** [review]

**Pull request merged**